pyugrid/grid_io/utils.py

```diff
@@ -1,128 +1,109 @@
 #!/usr/bin/env python
 
 """
-utilities to help with io
+Utilities to help with grid io
 
 NOTE: this isn't used yet, but should be useful for loading non
-UGRID-compliant files
+UGRID-compliant files.
+
 """
 
 from __future__ import (absolute_import, division, print_function)
 
-import numpy as np
 import netCDF4
-import pyugrid
+import numpy as np
+
+from ..ugrid import UGrid
 
 
 def load_from_varnames(filename, names_mapping, attribute_check=None):
     """
-    load a UGrid from a netcdf file where the roles are defined by the
-    names of the variables
+    Load a UGrid from a netcdf file where the roles are defined by the
+    names of the variables.
 
-    :param filename: the names of the file to load (or opendap url)
+    :param filename: names of the file to load (or OPeNDAP URL).
 
-    :param names_mapping: dict that maps the variable names to the UGRid components
+    :param names_mapping: dict that maps the variable names to UGrid components
 
     :param attribute_check=None: list of global attributes that are expected
     :type attribute_check: list of tuples to check. Example:
                            [('grid_type','triangular'),] will check if the
                            grid_type attribute is set to "triangular"
 
-    The names_mapping dict has to contain at least:
-
-    'nodes_lon', 'nodes_lat'
-
-    Optionally (and mostly required), it can contain:
-
-    'faces', 'face_face_connectivity', 'face_coordinates_lon', 'face_coordinates_lat'
+    The names_mapping dict has to contain at least: 'nodes_lon', 'nodes_lat'
 
+    Optionally (and mostly required), it can contain: face_face_connectivity',
+    'face_coordinates_lon', 'face_coordinates_lat', and 'faces'
 
     """
+    ug = UGrid()
     attribute_check = {} if attribute_check is None else attribute_check
 
-    # open the file
     nc = netCDF4.Dataset(filename)
 
-    # check if it's the file type we are looking for
-
+    # Check for the specified attributes.
     for name, value in attribute_check:
         if nc.getncattr(name).lower() != value:
             raise ValueError('This does not appear to be a valid file:\n'
                              'It does not have the "{}"="{}"'
                              'global attribute set'.format(name, value))
-    # create an empty UGrid:
-    ug = pyugrid.UGrid()
 
-    # load the nodes -- this is required
-    # nodes are usually stored in two different arrays
+    # Nodes.
     lon = nc.variables[names_mapping['nodes_lon']]
     lat = nc.variables[names_mapping['nodes_lat']]
 
-    num_nodes = lon.shape[0]
+    num_nodes = lon.size
     ug.nodes = np.zeros((num_nodes, 2), dtype=lon.dtype)
     ug.nodes[:, 0] = lon[:]
     ug.nodes[:, 1] = lat[:]
 
-    # load the faces
+    # Faces.
     faces = nc.variables[names_mapping['faces']]
-    # does it need to be transposed?
-    # assume there are more than three triangles...
+    # FIXME: This logic assumes there are more than three triangles.
```

Comment:
The latest UGRID spec covers this, but this is specifically for non-UGRID compatible files. So maybe the user should be required to specify that they want a transpose if they need it?

Also, if we do do this check, might as well check for ==, rather than <= -- so two triangle grids will work: there are examples that size :-)